lib/client/Client.js

```diff
@@ -90,6 +90,52 @@ export default class Client {
   }
 
   /**
+   * Implement the 2-step JMAP authentication protocol.<br />
+   * This method abstract the two authentication steps:
+   *
+   * 1. query the JMAP server to get a continuation token
+   * 2. query the JMAP server with the continuation token (and password), to get the final accessToken.
+   *
+   * @param username {String} The username to authenticate with
+   * @param deviceName {String} A unique device name
+   * @param continuationCallback {Function} A function that takes an {@link AuthContinuation}
+   *   object as argument, and should return a promise, that will eventually resolve with an
+   *   object denoting the chosen authentication method and the optional password (if method == password).
+   *
+   * @return {Promise} A {@link Promise} that will eventually be resovled with a {@link AuthAccess} object
+   */
+  authenticate(username, deviceName, continuationCallback) {
+    var authContinuation;
+    return this.transport
+      .post(this.authenticationUrl, this._defaultNonAuthenticatedHeaders(), {
+        username: username,
+        deviceName: deviceName,
+        clientName: 'jmap-client (https://github.com/linagora/jmap-client)',
+        clientVersion: Constants.VERSION
+      })
+      .then((data) => {
+        authContinuation = new AuthContinuation(data);
+        return authContinuation;
+      })
+      .then((authContinuation) => continuationCallback(authContinuation))
+      .then((continueData) => {
+        if (authContinuation.methods.indexOf(continueData.method) < 0) {
+          throw new Error('The "' + continueData.method + '" authentication protocol is not supported by the server.');
+        }
+        let param = {
+          token: continueData.token || authContinuation.continuationToken,
```

Comment:
I'd remove that the "continuationCallback" can provide the continuationToken, there is no reason for the caller to change it. Now that the authContinuation is kept locally, we should always take the CT from it.

Moreover this "||" branch is not tested.